node/state/state.go

```diff
@@ -25,12 +26,22 @@ func Components(coreEndpoint string, cfg key.Config) fx.Option {
 }
 
 // Service constructs a new state.Service.
-func Service(ctx context.Context, lc fx.Lifecycle, accessor state.Accessor, fsub fraud.Subscriber) *state.Service {
+func Service(ctx context.Context, lc fx.Lifecycle, accessor state.Accessor, fservice fraud.Service) *state.Service {
 	serv := state.NewService(accessor)
 	lc.Append(fx.Hook{
-		OnStart: serv.Start,
-		OnStop:  serv.Stop,
+		OnStart: func(context.Context) error {
+			proofs, err := fservice.Get(ctx, fraud.BadEncoding)
+			switch err {
+			case nil:
+				return fraud.ErrFraudExists{Type: fraud.BadEncoding, Proof: proofs}
+			case datastore.ErrNotFound:
+			default:
+				return err
```

Comment:
You can hide it inside Subscribe to make things DRY

Reply:
done